cap-ci/pipeline.template

```diff
@@ -1,21 +1,23 @@
 ---
+{{ $config := ( tmpl.Exec "config" ) | data.JSON}}
 
 groups:
 {{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
 - name: {{$Backend}}
   jobs:
-{{ range $_, $Jobs := slice "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
+
+{{ range $_, $Jobs := slice "create-k8s" "deploy" "smoke-tests" "cf-acceptance-tests-brain" "cf-acceptance-tests" "sync-integration-tests" "upgrade" "stratos" -}} # for jobs
```

Comment:
it won't fail. it will point out "create-k8s" as a job even if its set to "false". I did not change it as I have no idea why we need a list of jobs in the first place.

Reply:
same issue will happen after modularisation when we turn off some other job, eg. cats in pre-upgrades. Either we list all job including the ones which are `OFF` or we just stop listing the jobs, as it provides no value on UI front.

Reply:
We will get the final picture after  modularisation work.

Reply:
> it won't fail. it will point out "create-k8s" as a job even if its set to "false". I did not change it as I have no idea why we need a list of jobs in the first place.

Trying to deploy the pipeline afer disabling the create-k8s job will fail because there's no such job:

```
apply configuration? [yN]: y
error: invalid pipeline config:
invalid groups:
        group 'caasp4' has unknown job 'create-k8s-diego-caasp4-sa'
        group 'caasp4' has unknown job 'create-k8s-diego-caasp4-ha'
        group 'caasp4' has unknown job 'create-k8s-diego-caasp4-all'
        group 'caasp4' has unknown job 'create-k8s-eirini-caasp4-sa'
        group 'caasp4' has unknown job 'create-k8s-eirini-caasp4-ha'
        group 'caasp4' has unknown job 'create-k8s-eirini-caasp4-all'
        group 'aks' has unknown job 'create-k8s-diego-aks-sa'
        group 'aks' has unknown job 'create-k8s-diego-aks-ha'
        group 'aks' has unknown job 'create-k8s-diego-aks-all'
        group 'aks' has unknown job 'create-k8s-eirini-aks-sa'
        group 'aks' has unknown job 'create-k8s-eirini-aks-ha'
        group 'aks' has unknown job 'create-k8s-eirini-aks-all'
        group 'gke' has unknown job 'create-k8s-diego-gke-sa'
        group 'gke' has unknown job 'create-k8s-diego-gke-ha'
        group 'gke' has unknown job 'create-k8s-diego-gke-all'
        group 'gke' has unknown job 'create-k8s-eirini-gke-sa'
        group 'gke' has unknown job 'create-k8s-eirini-gke-ha'
        group 'gke' has unknown job 'create-k8s-eirini-gke-all'
        group 'eks' has unknown job 'create-k8s-diego-eks-sa'
        group 'eks' has unknown job 'create-k8s-diego-eks-ha'
        group 'eks' has unknown job 'create-k8s-diego-eks-all'
        group 'eks' has unknown job 'create-k8s-eirini-eks-sa'
        group 'eks' has unknown job 'create-k8s-eirini-eks-ha'
        group 'eks' has unknown job 'create-k8s-eirini-eks-all'
        group 'ALL' has unknown job 'create-k8s-diego-caasp4-sa'
        group 'ALL' has unknown job 'create-k8s-diego-caasp4-ha'
        group 'ALL' has unknown job 'create-k8s-diego-caasp4-all'
        group 'ALL' has unknown job 'create-k8s-eirini-caasp4-sa'
        group 'ALL' has unknown job 'create-k8s-eirini-caasp4-ha'
        group 'ALL' has unknown job 'create-k8s-eirini-caasp4-all'
        group 'ALL' has unknown job 'create-k8s-diego-aks-sa'
        group 'ALL' has unknown job 'create-k8s-diego-aks-ha'
        group 'ALL' has unknown job 'create-k8s-diego-aks-all'
        group 'ALL' has unknown job 'create-k8s-eirini-aks-sa'
        group 'ALL' has unknown job 'create-k8s-eirini-aks-ha'
        group 'ALL' has unknown job 'create-k8s-eirini-aks-all'
        group 'ALL' has unknown job 'create-k8s-diego-gke-sa'
        group 'ALL' has unknown job 'create-k8s-diego-gke-ha'
        group 'ALL' has unknown job 'create-k8s-diego-gke-all'
        group 'ALL' has unknown job 'create-k8s-eirini-gke-sa'
        group 'ALL' has unknown job 'create-k8s-eirini-gke-ha'
        group 'ALL' has unknown job 'create-k8s-eirini-gke-all'
        group 'ALL' has unknown job 'create-k8s-diego-eks-sa'
        group 'ALL' has unknown job 'create-k8s-diego-eks-ha'
        group 'ALL' has unknown job 'create-k8s-diego-eks-all'
        group 'ALL' has unknown job 'create-k8s-eirini-eks-sa'
        group 'ALL' has unknown job 'create-k8s-eirini-eks-ha'
        group 'ALL' has unknown job 'create-k8s-eirini-eks-all'
```

Reply:
> I did not change it as I have no idea why we need a list of jobs in the first place.

The jobs listing create the ("caasp4", "eks", "aks", "gke", "all") tabs.

>  Either we list all job including the ones which are `OFF` or we just stop listing the jobs, as it provides no value on UI front.

They do provide value, as I said, the jobs listing create the ("caasp4", "eks", "aks", "gke", "all") tabs.

Reply:
> We will get the final picture after modularisation work. As right now the job list is hardcoded, so it will give full list of jobs irrespective of`ON` or `OFF`. If we decide to keep populating jobs, then they will have to get generated via config.

Then it seems out of the scope for this PR.

Reply:
> > it won't fail. it will point out "create-k8s" as a job even if its set to "false". I did not change it as I have no idea why we need a list of jobs in the first place.
> 
> Trying to deploy the pipeline afer disabling the create-k8s job will fail because there's no such job:
> 
> ```
> apply configuration? [yN]: y
> error: invalid pipeline config:
> invalid groups:
>         group 'caasp4' has unknown job 'create-k8s-diego-caasp4-sa'

> ```

My apologies.
For now all I can do is turn off groups when create-k8s is `true`, as the list of jobs is hardcoded. We will have to wait on  modularisation work, so that the list of jobs is generated from configs for only jobs which are `ON`. This way existing functionality won't get affected. 
https://github.com/SUSE/cap/pull/6/commits/60923bcd93e774d86e45c02ff9d3cbd1ebb40200

Reply:
Honest thanks for reviewing this.

But, instead of adding a functionality that is broken, I would remove it then, until we have a commit that works. If not people are gonna set `create_k8s: false` and it's gonna explode on them. And we may forget about it.

Reply:
https://github.com/SUSE/cap/pull/6#issuecomment-619213471